opensquirrel/ir.py

```diff
@@ -247,6 +247,8 @@ def get_qubit_operands(self) -> list[Qubit]:
 
 
 class Gate(Statement, ABC):
+    _significant_digits_repr = 5
```

Comment:
Ideally, I would have the `_significant_digits_repr` variable just at `__repr__` scope, and not as a Gate member. However, I see it's used in the subclasses. I would:
1. Change `_round` to receive a `decimals=self.repr_decimals`, just as `np.round` [does](https://numpy.org/doc/stable/reference/generated/numpy.round.html), and then call `np.round(value, decimals)`. The only reason to keep `_round` and not to use `np.round` directly is that `_round` manages other types.
2. Change `_significant_digits_repr` to `repr_decimals` (as in, repr.decimals, or decimals used in `__repr__`). Just to be consistent with `np.round` naming. I think it's simpler and clearer.

Reply:
I've moved the helperfunction `_repr_round` and variable `_REPR_DECIMALS_` (used by the BSR and MatrixGate) to the global scope.

Reply:
Can you please remove the leading underscore? I would use it as a hint to the reader that a class member/method is intended to be private, but not here.